src/components/Select/Select.tsx

```diff
@@ -135,7 +137,7 @@ export const Select: React.FC<SelectProps> = ({
       ref={selectRef}
       {...rest}
     >
-      <input id={idRef} name={idRef} type='hidden' value={state.value} />
+      <input id={idRef} name={name || idRef} type='hidden' value={state.value} />
```

Comment:
I would not fall back to `idRef` here.
```suggestion
      <input id={idRef} name={name} type='hidden' value={state.value} />
```